### PR TITLE
[UNR-1641] - Spatial Position not updating for AI Characters

### PR DESCRIPTION
**Pull request by jnicholas-io**
#### Description
AI Characters were not having their Spatial position updated in the inspector, making it difficult to know which entity in the inspector corresponded to which Character in game. Actor replication in SpatialActorChannel would not update an Actor's Spatial position if it had a parent and the Actor itself was not an `APlayerController`.

Fixed by adding an extra check to `UpdateSpatialPosition`, if the replicating Actor has an owner and that owner is replicated, return early as before since the owner will be replicated and have it's Spatial position updated along with all of its children. In the case that the replicating Actor's owner is *not* replicated, continue and update its Spatial position

#### Release note
Fixed issue where an Actor's Spatial position was not updated if it had an owner that was not replicated (eg. AI Character possessed by an AIController)

#### Primary reviewers
@mattyoung-improbable @improbable-valentyn

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 1]
```diff
@@ -892,9 +892,16 @@ void USpatialActorChannel::UpdateSpatialPosition()
...
 	// we want to defer updating position until we reach the highest parent.
-	if ((Actor->GetOwner() != nullptr || Actor->GetNetConnection() != nullptr) && !Actor->IsA<APlayerController>())
+	AActor* ActorOwner = Actor->GetOwner();
+
+	if ((ActorOwner != nullptr || Actor->GetNetConnection() != nullptr) && !Actor->IsA<APlayerController>())
 	{
-		return;
+		// If this Actor's owner is not replicated (eg. parent = AI Controller), the actor will not have it's spatial
+		// position updated as this code will never be run for the parent. 
+		if (ActorOwner != nullptr && ActorOwner->GetIsReplicated())
```
Comment (tenevdev) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: This means the check for `Actor->GetNetConnection()` no longer matters and I believe it probably did. Could you expand on why this check was needed in the first place and remove it or add an extra case?

Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: This should only let it through if it has an owner that isn't replicated.
```suggestion
		if (!(ActorOwner != nullptr && !ActorOwner->GetIsReplicated()))
```

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 2]
```diff
@@ -892,9 +892,16 @@ void USpatialActorChannel::UpdateSpatialPosition()
 	// If this Actor is a PlayerController, we want to update all of its children and its possessed Pawn.
 	// That means if this Actor has an Owner or has a NetConnection and is NOT a PlayerController
 	// we want to defer updating position until we reach the highest parent.
-	if ((Actor->GetOwner() != nullptr || Actor->GetNetConnection() != nullptr) && !Actor->IsA<APlayerController>())
+	AActor* ActorOwner = Actor->GetOwner();
+
+	if ((ActorOwner != nullptr || Actor->GetNetConnection() != nullptr) && !Actor->IsA<APlayerController>())
 	{
-		return;
+		// If this Actor's owner is not replicated (eg. parent = AI Controller), the actor will not have it's spatial
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]:
```suggestion
		// If this Actor's owner is not replicated (e.g. parent = AI Controller), the actor will not have it's spatial
```

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp [hunk 3]
```diff
@@ -892,9 +892,16 @@ void USpatialActorChannel::UpdateSpatialPosition()
...
+		// position updated as this code will never be run for the parent. 
+		if (!(ActorOwner != nullptr && !ActorOwner->GetIsReplicated()))
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]:
```suggestion
		if (!(Actor->GetNetConnection() == nullptr && ActorOwner != nullptr && !ActorOwner->GetIsReplicated()))
```

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp (hunk 1 above)
Comment (jnicholas-io) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp]: fixed

**Pull request merged**